src/Toolkit/Toolkit.Maui/FeatureForm/BarcodeImageAnalyzer.Android.cs

```diff
@@ -0,0 +1,175 @@
+﻿#if ANDROID
+using Android.Gms.Tasks;
+using Android.Graphics;
+using Android.Runtime;
+using AndroidX.Camera.Core;
+using Microsoft.Maui.Graphics.Platform;
+using Xamarin.Google.MLKit.Vision.Barcode.Common;
+using Xamarin.Google.MLKit.Vision.BarCode;
+using RectF = Android.Graphics.RectF;
+using Size = Android.Util.Size;
+
+namespace Esri.ArcGISRuntime.Toolkit.Maui
+{
+
+    internal sealed class BarcodeInfo
+    {
+        public RectF BoundingBox { get; }
+        public string RawValue { get; }
+        public long LastSeenFrame { get; }
+
+        public BarcodeInfo(RectF boundingBox, string rawValue, long lastSeenFrame)
+        {
+            BoundingBox = boundingBox;
+            RawValue = rawValue;
+            LastSeenFrame = lastSeenFrame;
+        }
+    }
+
+    internal sealed class BarcodeImageAnalyzer : Java.Lang.Object, ImageAnalysis.IAnalyzer
+    {
+        private readonly Action<List<BarcodeInfo>> _onSuccess;
+        private Matrix? _sensorToTargetMatrix;
+        private long _frames;
+        private readonly Dictionary<string, BarcodeInfo> _barcodeMap = new();
+        private readonly IBarcodeScanner _barcodeScanner;
+
+        public BarcodeImageAnalyzer(Action<List<BarcodeInfo>> onSuccess)
+        {
+            _onSuccess = onSuccess;
+            _barcodeScanner = BarcodeScanning
+                                .GetClient(new BarcodeScannerOptions.Builder()
+                                             .EnableAllPotentialBarcodes()
+                                             .Build());
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            base.Dispose(disposing);
+            if (disposing)
+                _barcodeScanner.Dispose();
+        }
+
+        Size ImageAnalysis.IAnalyzer.DefaultTargetResolution => new(1920, 1080);
+
+        int ImageAnalysis.IAnalyzer.TargetCoordinateSystem => ImageAnalysis.CoordinateSystemViewReferenced;
+
+        void ImageAnalysis.IAnalyzer.Analyze(IImageProxy image)
+        {
+            var mediaImage = image.Image;
+            if (mediaImage != null)
+            {
+                _barcodeScanner.Process(mediaImage, image.ImageInfo.RotationDegrees)
+                    .AddOnSuccessListener(new OnSuccessListener(barcodes =>
+                    {
+                        _frames = (_frames + 1) % long.MaxValue;
```

Comment:
It's good to think ahead to the 9223372036854775808th frame 😀 

The wrapping around might make `.Where(info => info.LastSeenFrame < _frames - 4)` (in ProcessBarcodes below) behave oddly, but I'm not sure how realistic is this scenario.

Reply:
This is coming from the direct port of kotlin: https://github.com/Esri/arcgis-maps-sdk-kotlin-toolkit/blob/30b0339327181595db2c889eb33c7ce8f7914a8d/toolkit/featureforms/src/main/java/com/arcgismaps/toolkit/featureforms/internal/components/barcode/BarcodeImageAnalyzer.kt#L82